src/nvim/api/buffer.c

```diff
@@ -392,7 +395,9 @@ void nvim_buf_set_lines(uint64_t channel_id,
   // Only adjust marks if we managed to switch to a window that holds
   // the buffer, otherwise line numbers will be invalid.
   if (save_curbuf.br_buf == NULL) {
-    mark_adjust((linenr_T)start, (linenr_T)(end - 1), MAXLNUM, extra);
+    mark_adjust((linenr_T)start, (linenr_T)(end - 1), MAXLNUM, extra, false);
```

Comment:
Instead of adding `extmark_adjust` after a lot of (almost every?) `mark_adjust` calls, can't we add a `kExtmarkNoMove` variant for the cases extmark shouldn't be adjusted and then make `mark_adjust` take ExtmarkReverseType enum?

Otherwise the risk is, when futher `mark_adjust` calls will be added, extmark might be forgotten.